cohortextractor/generate_codelist_report.py

```diff
@@ -83,6 +91,31 @@ def events_query(start_date, end_date):
     """
 
 
+def practice_query(end_date):
+    # Note that this does not quite match patients.registered_practice_as_of(end_date),
+    # since a deceased patient might have been deregistered by the end date.
+    #
+    # In patients.registered_practice_as_of(end_date), the innermost SELECT query's
+    # WHERE clause would contain an extra `AND EndDate > {end_date}`.
+    return f"""
+    SELECT * INTO #practice FROM (
+        SELECT Patient_ID, t.Organisation_ID
+        FROM (
+            SELECT RegistrationHistory.Patient_ID, Organisation_ID,
+            ROW_NUMBER() OVER (
+                PARTITION BY RegistrationHistory.Patient_ID
+                ORDER BY StartDate DESC, EndDate DESC, Registration_ID
+            ) AS rownum
+            FROM RegistrationHistory
+            WHERE StartDate <= {end_date}
+        ) t
+        LEFT JOIN Organisation
```

Comment:
Do we need to join on organisation?

Reply:
Realised my confusion earlier was mostly because I hadn't seen the `WHERE t.rownum = 1`, so removing the join on organisation will make that much more obvious.

Is that being covered by the tests at the moment?

Reply:
Just to clarify my understanding, the reason for the practice query is that a patient may have been registered at more than one practice when we do `StartDate <= {end_date} AND EndDate > {end_date}` in `population_query()`. So the purpose of `practice_query()` is to get the most recent practice.

Would it be worth renaming `practice_query` to `most_recent_practice_query`, or if that's too much of a mouthful, to add a code comment to say why this is needed?